### PR TITLE
Upgrade vite version

### PR DESCRIPTION
**Pull request by calufornia**
## Description

This pulls in various fixes, in particular: https://github.com/vitejs/vite/issues/9986

## Test plan
n/a

### CODE REVIEW

File: pkg/build/package.json [hunk 1]
```diff
@@ -10,6 +10,6 @@
         "react": "18.2.0",
         "react-dom": "18.2.0",
         "typescript": "4.9.5",
-        "vite": "4.1.4"
+        "vite": "4.2.0"
```
Comment (calufornia) [inline pkg/build/package.json:13]: Why is this here?

File: pkg/build/views/package.json [hunk 2]
```diff
@@ -18,6 +18,6 @@
     "@types/object-hash": "3.0.2",
     "@vitejs/plugin-react": "3.1.0",
     "typescript": "4.9.5",
-    "vite": "4.1.4"
+    "vite": "4.2.0"
```
Comment (calufornia) [inline pkg/build/views/package.json:21]: Actually, why is this here?

**Review by leeweisberger — APPROVED** [review]
